src/validation/validator.rs

```diff
@@ -419,31 +407,34 @@ impl<
         &self,
         bundle_id: BundleId,
         bundle: &TransitionBundle,
-        pub_witness: XWitnessTx,
-        input_map: BTreeMap<OpId, BTreeSet<XOutpoint>>,
+        pub_witness: Tx,
+        input_map: BTreeMap<OpId, BTreeSet<Outpoint>>,
     ) {
-        let witness_id = pub_witness.witness_id();
-        for (vin, opid) in &bundle.input_map {
-            let Some(outpoints) = input_map.get(opid) else {
-                self.status
-                    .borrow_mut()
-                    .add_failure(Failure::BundleExtraTransition(bundle_id, *opid));
-                continue;
-            };
-            let layer1 = pub_witness.layer1();
-            let pub_witness = pub_witness.as_reduced_unsafe();
-            let Some(input) = pub_witness.inputs.get(vin.to_usize()) else {
-                self.status
-                    .borrow_mut()
-                    .add_failure(Failure::BundleInvalidInput(bundle_id, *opid, witness_id));
-                continue;
-            };
-            if !outpoints.contains(&XChain::with(layer1, input.prev_output)) {
-                self.status
-                    .borrow_mut()
-                    .add_failure(Failure::BundleInvalidCommitment(
-                        bundle_id, *vin, witness_id, *opid,
-                    ));
+        let witness_id = pub_witness.txid();
+        for (vin, opids) in &bundle.input_map {
+            for opid in opids {
+                if self.trusted_op_seals.contains(&opid) {
+                    continue;
+                }
+                let Some(outpoints) = input_map.get(&opid) else {
```

Comment:
Here, the `known_transitions` field of the bundle is not checked at all!

Instead, everything agains double-spent is relied onto a `self.trusted_op_seals`, injected as a backdoor from the application level!